src/WooCommerce.php

```diff
@@ -1063,12 +1063,14 @@ public static function getEarliestBackInStock(\WC_Product $product): string {
    * @implements woocommerce_de_get_deliverytime_string_label_string
    */
   public static function woocommerce_de_get_deliverytime_string_label_string($label_string, $product) {
-    if (!$product) {
+    // Avoid the execution of this function for product variants which can be
+    // evaluated over the parent product in the next method
+    if (!$product || $product instanceof \WC_Product_Variation) {
```

Comment:
```suggestion
    if (!$product) {
```
This change might interfere with other contexts where it shouldn't.